README.md

```diff
@@ -61,6 +61,46 @@ PyPI, which is recommended to restrict the access the action has.
 The secret used in `${{ secrets.PYPI_API_TOKEN }}` needs to be created on the
 settings page of your project on GitHub. See [Creating & using secrets].
 
+### Publishing with OpenID Connect
+
+**IMPORTANT**: This functionality is in beta, and will not work for you
+unless you're a member of the PyPI OIDC beta testers' group. For more
+information, see
+[warehouse#12965](https://github.com/pypi/warehouse/issues/12965).
+
+This action supports PyPI's
+[OpenID Connect publishing](https://pypi.org/help/#openid-connect)
+implementation, which allows authentication to PyPI without a manually
+configured API token or username/password combination. To perform
+OIDC publishing with this action, your project's OIDC publisher must
+already be configured on PyPI.
+
+To enter the OIDC flow, configure this action's job with the `id-token: write`
+permission and **without** an explicit username or password:
+
+```yaml
+jobs:
+  pypi-publish:
+    name: upload release to PyPI
+    runs-on: ubuntu-latest
+    permissions:
+      # IMPORTANT: this permission is mandatory for OIDC publishing
+      id-token: write
```

Comment:
I think it's better on the same line since it has a "this" pointer rather than talking about something on another line:
```suggestion
      id-token: write  # IMPORTANT: this permission is mandatory for OIDC publishing
```
Plus, comments tend to drift away from the lines they refer to over time, which is why it's considered better to have them on the same line whenever possible.